index.html

```
@@ -0,0 +1,306 @@
+
+            let tmp, j, i = list.length - 1
+
+            for (; i > 0; i -= 1) {
+                j = Math.floor(Math.random() * (i + 1));
+                tmp = list[i];
+                list[i] = list[j];
+                list[j] = tmp;
+            }
+
+            return result;
+        }
+
+
+        /**
+         * Remove any people that do not have the name we are
+         * searching for.
+         */
+        function filterByName(searchForName, personList) {
+            return personList.filter((person) => {
+                return person.firstName === searchForName || person.lastName === searchForName;
+            });
+        }
+
+
+        /**
+         * Takes in a property of an object list, e.g. "name" below
+         *
+         *    people = [{ name: 'Sam' }, { name: 'Jon' }, { name: 'Kevin' }]
+         *
+         * And returns a function that will sort that list, e.g.
+         *
+         *    const sortPeopleByName = sortObjListByProp('name');
+         *    const sortedPeople = sortPeopleByName(people);
+         *
+         *  We now have:
+         *
+         *    console.log(sortedPeople)
+         *    > [{ name: 'Jon' }, { name: 'Kevin' }, { name: 'Sam' }]
+         *
+         */
+        function sortObjListByProp(prop) {
+            return function(objList) {
+                // Make a copy & don't mutate the passed in list
+                let result = objList.slice(1);
+
+                result.sort((a, b) => {
+                    if (a[prop] < b[prop]) {
+                        return -1;
+                    }
+
+                    if (a[prop] > b[prop]) {
+                        return 1;
+                    }
+
+                    return 1;
+                });
+
+                return result;
+            };
+        }
+
+        const sortByFirstName = sortObjListByProp('firstName');
+
+        const sortByLastName = (personList) => sortByFirstName(personList).reverse();
```

This is just sorting by first name, and then reversing the order. The fix here is mirror "sortByFirstName", except pass the 'lastName' prop instead.